File: 0198-house-robber/0198-house-robber.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        
        if len(nums) < 3:
            return max(nums)
        
        
        
        for i in range(2, len(nums)):
            if i == 2:
                nums[2] += nums[0]
            else:
                nums[i] += max(nums[i-2], nums[i-3])

        return max(nums[-1], nums[-2])
        

File: 0198-house-robber/0198-house-robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        
        int first = 0;
        int second = 0;
        
        for (int i = 0; i < nums.size(); i++) {
            int temp = max(nums[i] + first, second);
            first = second;
            second = temp;
        }
        
        return second;
    }
};